src/prefect/orion/api/deployments.py

```diff
@@ -35,10 +37,20 @@ async def create_deployment(
     When upserting, any scheduled runs from the existing deployment will be deleted.
     """
 
-    # hydrate the input model into a full model
-    deployment = schemas.core.Deployment(**deployment.dict())
-
     async with db.session_context(begin_transaction=True) as session:
+        # hydrate the input model into a full model
+        deployment_dict = deployment.dict(
+            exclude={"worker_pool_name", "worker_pool_queue_name"}
+        )
+        if deployment.worker_pool_name and deployment.worker_pool_queue_name:
```

Comment:
do you mean `or` here?  A queue name is optional but a worker pool won't be

Reply:
actually, is the queue name even relevant here?

Reply:
A deployment is a associated with a pool queue and we need a pool name and a pool queue name in order to get an ID for a pool queue. If we are given just a pool name then it makes sense to associate the deployment with the default queue for the pool. I'll add logic for that case.